tests/tx/test_verification.py

```diff
@@ -0,0 +1,448 @@
+        verify_outputs_wrapped.assert_called_once()
+        verify_data_wrapped.assert_called_once()
+        verify_sigops_output_wrapped.assert_called_once()
+        verify_parents_wrapped.assert_called_once()
+        verify_height_wrapped.assert_called_once()
+
+        # MergeMinedBlock methods
+        verify_pow_wrapped.assert_called_once()
+
+    def test_transaction_verify_basic(self) -> None:
+        tx = self._get_valid_tx()
+
+        verify_parents_basic_wrapped = Mock(wraps=tx.verify_parents_basic)
+        verify_weight_wrapped = Mock(wraps=tx.verify_weight)
+        verify_pow_wrapped = Mock(wraps=tx.verify_pow)
+        verify_number_of_inputs_wrapped = Mock(wraps=tx.verify_number_of_inputs)
+        verify_outputs_wrapped = Mock(wraps=tx.verify_outputs)
+        verify_sigops_output_wrapped = Mock(wraps=tx.verify_sigops_output)
+
+        with (
+            patch.object(Transaction, 'verify_parents_basic', verify_parents_basic_wrapped),
+            patch.object(Transaction, 'verify_weight', verify_weight_wrapped),
+            patch.object(Transaction, 'verify_pow', verify_pow_wrapped),
+            patch.object(Transaction, 'verify_number_of_inputs', verify_number_of_inputs_wrapped),
+            patch.object(Transaction, 'verify_outputs', verify_outputs_wrapped),
+            patch.object(Transaction, 'verify_sigops_output', verify_sigops_output_wrapped),
+        ):
+            self.manager.verification_service.verify_basic(tx)
+
+        # Transaction methods
+        verify_parents_basic_wrapped.assert_called_once()
+        verify_weight_wrapped.assert_called_once()
+        verify_pow_wrapped.assert_called_once()
+        verify_number_of_inputs_wrapped.assert_called_once()
+        verify_outputs_wrapped.assert_called_once()
+        verify_sigops_output_wrapped.assert_called_once()
+
+    def test_transaction_verify_without_storage(self) -> None:
+        tx = self._get_valid_tx()
+
+        verify_pow_wrapped = Mock(wraps=tx.verify_pow)
+        verify_number_of_inputs_wrapped = Mock(wraps=tx.verify_number_of_inputs)
+        verify_outputs_wrapped = Mock(wraps=tx.verify_outputs)
+        verify_sigops_output_wrapped = Mock(wraps=tx.verify_sigops_output)
+
+        with (
+            patch.object(Transaction, 'verify_pow', verify_pow_wrapped),
+            patch.object(Transaction, 'verify_number_of_inputs', verify_number_of_inputs_wrapped),
+            patch.object(Transaction, 'verify_outputs', verify_outputs_wrapped),
+            patch.object(Transaction, 'verify_sigops_output', verify_sigops_output_wrapped),
+        ):
+            tx.verify_without_storage()
+
+        # Transaction methods
+        verify_pow_wrapped.assert_called_once()
+        verify_number_of_inputs_wrapped.assert_called_once()
+        verify_outputs_wrapped.assert_called_once()
+        verify_sigops_output_wrapped.assert_called_once()
+
+    def test_transaction_verify(self) -> None:
```

Comment:
Maybe we should open a KTLO to rename `Transaction.verify_unsigned_skip_pow()` to something else as it's not part of the verification process. I checked that it is used by an HTTP Endpoint so we might move the code to this specific endpoint.

Reply:
Created in https://github.com/HathorNetwork/hathor-core/issues/825. I can do that as soon as the refactors are merged.

Reply:
Thanks!